### PR TITLE
Add onCheckCellIsEditable event to cell

### PR DESCRIPTION
**Pull request by diegomurakami**
## Description
Expose a pre-cell-edit event

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/adazzle/react-data-grid/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
There is no pre-edit event.

**What is the new behavior?**
A new event check will be executed before editing a cell.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/ReactDataGrid.js [hunk 1]
```diff
@@ -759,11 +760,18 @@ const ReactDataGrid = React.createClass({
...
-    let col = this.getColumn(idx);
+    let column = this.getColumn(idx);
 
-    if (ColumnUtils.canEdit(col, row, this.props.enableCellSelect) && !this.isActive()) {
+    if (ColumnUtils.canEdit(column, row, this.props.enableCellSelect) && !this.isActive()) {
       let selected = Object.assign(this.state.selected, {idx: idx, rowIdx: rowIdx, active: true, initialKeyCode: keyPressed});
-      this.setState({selected: selected}, this.scrollToColumn(idx));
+      let showEditor = true;
+      if (typeof this.props.onCellSelect === 'function') {
+        let args = Object.assign({ row, column }, selected);
```
Comment (malonecj) [inline src/ReactDataGrid.js]: ```let args = Object.assign({}, { row, column }, selected);```

File: src/ReactDataGrid.js [hunk 2]
```diff
@@ -759,11 +760,18 @@ const ReactDataGrid = React.createClass({
...
+        showEditor = this.props.onCellSelect(args);
```
Comment (malonecj) [inline src/ReactDataGrid.js]: Add to prop types

```onCheckCellIsEditable /*called before cell is set active, returns a boolean to determine whether cell is editable*/```

File: src/ReactDataGrid.js [hunk 3]
```diff
@@ -103,7 +103,8 @@ const ReactDataGrid = React.createClass({
     onGridKeyUp: React.PropTypes.func,
     onGridKeyDown: React.PropTypes.func,
     rowGroupRenderer: React.PropTypes.func,
-    rowActionsCell: React.PropTypes.func
+    rowActionsCell: React.PropTypes.func,
+    onCellSelect: React.PropTypes.func
```
Comment (malonecj) [inline src/ReactDataGrid.js]:
```javascript
onCheckCellIsEditable:  React.PropTypes.func 
/*called before cell is set active, returns a boolean to determine whether cell is editable*/
```

**Comment by diegomurakami** [comment]
@malonecj All done ;)

**Pull request merged**